Doc/library/subprocess.rst

```diff
@@ -654,6 +655,17 @@ functions.
    .. versionadded:: 3.10
       The ``pipesize`` parameter was added.
 
+   *read_callback* can be used to replace the default internal handler for
```

Comment:
I'd rather not mention internal implementation details in the public docs.  Instead, a wording more like:

```
*read_callback*, if supplied, is called upon :data:`PIPE` data being read
for either *stdout* or *stderr* of the child process. It must expect four args:
The :class:``Popen`` instance, an ``is_stdout`` (bool), an ``output_buffer``
list of bytes to append accumulated data to, and the ``data`` (bytes) just read
from the pipe. In order to maintain standard Popen behavior of collecting
pipe data to be available at the finish in *stdout* and *stderr* attributes,
``output_buffer.append(data)`` must be called within the callback.
```

Or something along those lines. It is more friendly. No mention of internal things.

Yet even writing that... the way this reads, it suggests the API for the callback really should be different. The arguments are not supplied in the order they need to be discussed.  How about:

`(self: subprocess.Popen, data: bytes, output_list: list[bytes])`

as the callback API.

I'm still torn on the *is_stdout* bool. It is annoying to see a bool. It means many implementations also need to have a conditional on it.  I think we should have two callbacks instead.  read_stdout_callback & read_stderr_callback.  People who want the same behavior on both just pass both args pointing to the same callable.  People who want different behavior supply different callbacks.

Reply:
Done.